keps/sig-network/3458-remove-transient-node-predicates-from-service-controller/README.md

```diff
@@ -0,0 +1,337 @@
+allowed to drain in those cases, even though the load balancer might support
+that. Moreover: on large clusters with a lot nodes and entropy, re-syncing load
+balancers like this can lead to rate-limiting by the cloud provider due to an
+excessive amount of update calls.
+
+As to enable connection draining, reduce cloud provider API calls and simplify
+the KCCMs sync loop: this KEP proposes that the service controller stops
+synchronizing the load balancer node set in these cases. Seeing as how this has
+always been the case, a new feature gate `LoadBalancerConnectionDraining` will
+be introduced, which will be used to enable the more optimal behavior.
+
+## Motivation
+
+Abruptly terminating connections in the cases defined by a) and b) above can be
+seen as buggy behavior and should be improved. By enabling connection draining,
+applications are allowed profit from graceful shutdown / termination, for what
+concerns cluster ingress connectivity. Users of Kubernetes will also see a
+reduction in the amount of cloud API calls, for what concerns calls stemming
+from syncing load balancers with the Kubernetes cluster state.
+
+### Goals
+
+- Stop re-configuring the load balancers' node set for cases a) and b) above
+
+### Non-Goals
+
+- Stop re-configuring the load balancers' node set for fully deleted /
+  newly added cluster nodes, or for nodes which get annotated with
+  `node.kubernetes.io/exclude-from-external-load-balancers`.
+
+## Proposal
+
+### Risks and Mitigations
+
+#### Risk
+
+- Cloud providers which do not allow VM deletion when the VM is referenced by
+  other constructs, will block the cluster auto-scaler (CA) from deleting the VM
+  upon downscale. This will result in reduced downscale performance by the CA,
+  or completely block VM deletion from happening - this is because the service
+  controller will never proceed to de-reference the VM from the load balancer
+  node set until the Node is fully deleted in the API server, which will never
+  occur until the VM is deleted. The three major cloud providers (GCP/AWS/Azure)
+  do however support this, and it is not expected that other providers don't.
+- By removing the b) predicate above we are delaying the removal of the node
+  from the load balancers' node set until the node is completely deleted in the
+  API server. Workloads relying on this behavior will observe a change in
+  behavior, but this also not expected.
+
+#### Mitigations
+
+- Cloud providers/workloads which do not support the behavior mentioned in
+  (Risk)[#risk], have the possibility to set the feature flag to false, thus
+  default back to the current mechanism.
+
+## Design Details
+
+- Implement the change in the service controller and ensure it does not add /
+  remove nodes from the load balancers' node set for cases a) and b) mentioned
+  in (Summary)[#Summary]
+- Add the feature gate: `LoadBalancerConnectionDraining`, set it to "on" by
+  default and promote it directly to GA
```

Comment:
Let's just call it Beta (since most GA's are "lockedToDefault") in 27 and GA in 28 or 29?

Reply:
OK, will update